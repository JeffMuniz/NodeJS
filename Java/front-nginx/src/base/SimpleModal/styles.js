import styled from "styled-components";
import { lighterBlack } from "@colors";

export const ModalWrapper = styled.div`
  position: fixed;
  z-index: 9;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  min-width: 100vw;
  min-height: 100vh;
  background-color: rgba(0, 0, 0, 0.4);
`;

export const ModalContainer = styled.div`
  width: ${props => props.width || "600px"};
  max-width: 80vw;
  min-height: ${props => props.height || "352px"};
  background-color: white;
  overflow: none;
  box-shadow: 10px 10px 20px rgba(0, 0, 0, 0.2);
  border-radius: 5px;
`;

export const ModalHeader = styled.div`
  position: relative;
  float: left;
  width: 100%;
  height: 0;
`;

export const ModalTitle = styled.div`
  float: left;
  padding: 40px;
  font-family: "Open Sans", sans-serif;
  font-size: 28px;
  color: ${lighterBlack};
`;

export const ModalCloseButton = styled.div`
  position: relative;
  float: right;
  right: 20px;
  top: 20px;
`;

export const ModalCloseIcon = styled.div`
  cursor: pointer;
`;

export const ModalContent = styled.div`
  padding-left: ${({ paddingless }) => !paddingless && "40px"};
  padding-right: ${({ paddingless }) => !paddingless && "40px"};
  padding-bottom: ${({ paddingless }) => !paddingless && "40px"};
  padding-top: ${({ hasTitle, paddingless }) =>
    !hasTitle && !paddingless && "40px"};
  margin-top: ${({ hasTitle }) => hasTitle && "120px"};
  overflow: auto;
  max-height: 100vh;
`;
