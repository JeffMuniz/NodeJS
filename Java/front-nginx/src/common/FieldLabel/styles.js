import styled from "styled-components";
import { lighterBlack } from "@colors";

export const FieldLabelWrapper = styled.div`
  width: ${props => props.width};
  min-width: ${props => props.minWidth};
  padding-right: 10px;
`;

export const Label = styled.div`
  line-height: 22px;
  font-size: 12px;
  color: ${lighterBlack};
`;

export const Value = styled.div`
  line-height: 26px;
  font-size: 16px;
  color: ${lighterBlack};
  font-weight: bold;
`;
