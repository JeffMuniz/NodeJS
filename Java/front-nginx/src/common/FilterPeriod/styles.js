import styled from "styled-components";

export const FilterPeriodWrapper = styled.div`
  display: flex;
`;

export const FilterLabel = styled.div`
  margin-right: 5px;
`;
