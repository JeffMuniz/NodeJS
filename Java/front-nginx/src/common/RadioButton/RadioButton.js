import React from "react";
import PropTypes from "prop-types";

import {
  TextError,
  Description,
  Wrapper,
  Label,
  Box,
  SpanBox,
} from "./RadioButton.styles";

const RadioButton = props => {
  const {
    id,
    selectedValue,
    error,
    onChange,
    labelBefore,
    options,
    name,
    isDisabled,
    setFieldValue,
    setFieldTouched,
    values,
  } = props;

  if (!Array.isArray(options)) return null;

  return (
    <Wrapper>
      {options.map(option => (
        <Label htmlFor={`${id}_${option.id}`} key={option.id}>
          {labelBefore && option.label && (
            <Description labelBefore={labelBefore}>{option.label}</Description>
          )}
          <Box
            id={`${id}_${option.id}`}
            type="radio"
            name={name}
            value={values.option}
            checked={selectedValue === option.value}
            onChange={() => {
              setFieldValue(name, option.value);
              onChange(option.value);
            }}
            disabled={isDisabled}
            onBlur={() => setFieldTouched(name)}
          />
          <SpanBox />
          {!labelBefore && option.label && (
            <Description labelBefore={labelBefore}>{option.label}</Description>
          )}
        </Label>
      ))}
      {error && <TextError id={`${id}_error`}>{error}</TextError>}
    </Wrapper>
  );
};

RadioButton.propTypes = {
  selectedValue: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
  ]),
  name: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
        PropTypes.bool,
      ]),
      id: PropTypes.string,
      label: PropTypes.string,
    }),
  ).isRequired,
  isDisabled: PropTypes.bool,
  labelBefore: PropTypes.bool,
  error: PropTypes.string,
  id: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  setFieldValue: PropTypes.func.isRequired,
  setFieldTouched: PropTypes.func.isRequired,
  values: PropTypes.shape({
    option: PropTypes.string,
  }).isRequired,
};

RadioButton.defaultProps = {
  labelBefore: true,
  selectedValue: "",
  isDisabled: false,
  error: null,
};

export default RadioButton;
