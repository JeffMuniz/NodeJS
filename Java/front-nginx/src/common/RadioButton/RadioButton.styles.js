import styledWeb from "styled-components";

import {
  white,
  darkWhite,
  lighterGreen,
  simpleDarkenGrey,
  veryLightBlack,
} from "src/styles/colors";

export const SpanBox = styledWeb.span`
  position: absolute;
  top: 3px;
  height: 15px
  width: 15px
  border: 1px solid ${simpleDarkenGrey};
  background-color: ${darkWhite};
  border-radius: 50%;
`;

export const Box = styledWeb.input`
  position: absolute;
  opacity: 0;
  cursor: pointer;

  &:checked ~ ${SpanBox} {
    background-color: ${white};

    &:after {
      content: "";
      position: absolute;
      top: 4px;
      left: 4px;
      width: 7px;
      height: 7px;
      border-radius: 50%;
      background: ${lighterGreen};
    }
  }
`;

export const TextError = styledWeb.span`
  color: red;
  display: block;
  font-style: italic;
  font-size: 12px;
`;

export const Description = styledWeb.span`
  color: ${veryLightBlack};
  font-size: 14px;
  font-weight: 600;
  line-height: 24px;
  margin: ${props => (props.labelBefore ? "0 5px 0 0" : "0 0 0 30px")};
`;

export const Label = styledWeb.label`
  position: relative;
  margin-left: 40px
  :nth-child(1) {
    margin-left: 0;
  }
`;

export const Wrapper = styledWeb.div`
  margin: 10px 0;
  width: 100%;
`;
