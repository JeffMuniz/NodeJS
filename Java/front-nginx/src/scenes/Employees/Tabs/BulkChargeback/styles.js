import styled from "styled-components";

import { TableCol, TableHeaderCol as BaseTableHeaderCol } from "@base";
import { blue, shark } from "@colors";

export const RouterLink = styled.p`
  font-size: 14px;
  font-weight: bold;
  line-height: 19px;
  text-decoration: none;
  color: ${blue};
  cursor: pointer;
`;

export const Text = styled(TableCol)`
  &&& {
    word-break: break-all;
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
    font-size: 14px;
    line-height: 19px;
    color: ${props => props.fill || shark};
    padding-left: 0;
    padding-right: 0;
  }
`;

export const TableHeaderCol = styled(BaseTableHeaderCol)`
  &&& {
    padding-left: 0;
    padding-right: 0;
  }
`;
