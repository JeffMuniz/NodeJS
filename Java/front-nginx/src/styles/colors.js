export const black = "#000000";
export const lightBlack = "#222222";
export const lighterBlack = "#272727";
export const darkBlack = "#333333";
export const veryLightBlack = "#3F4142";
export const shark = "#2e2f30";
export const mineShaft = "#333333";

export const white = "#FFFFFF";
export const darkWhite = "#F8F8F8";
export const darkerWhite = "#FBFBFB";
export const whiteF2 = "#F2F2F2";
export const seashell = "#F1F1F1";

export const red = "#FF053F";
export const redFlagGetnet = "#FF0000";
export const darkRed = "#ED5050";
export const redFlagSantander = "#CC2700";
export const persianRed = "#CC3030";
export const wildWatermelon = "#FF5B68";

export const persianPlum = "#721c24";
export const cherub = "#f8d7da";
export const azalea = "#f5c6cb";

export const blue = "#009CFF";
export const lightBlue = "#8FD3FF";

export const purple = "#BA08D2";
export const darkPurple = "#4e2789";

export const green = "#15C759";
export const lightGreen = "#BA08D2";
export const lighterGreen = "#40E18B";
export const darkGreen = "#2AD178";

export const progressBarGrey = "#929495";
export const darkenGrey = "#9B9B9B4C";
export const simpleDarkenGrey = "#9B9B9B";
export const darkGrey = "#EBEBEB";
export const grey = "#666666";
export const ligthGrey = "#D8D8D819";
export const veryLigtherGrey = "#D9D9D9";
export const gainsboro = "#D8D8D8";
export const ligtherGrey = "#7373734C";
export const navBarGrey = "#393B3B";
export const borderGrey = "#E7E7E7";
export const dustyGray = "#979797";
export const disabledGrey = "#E1E1E1";
export const disabledDarkGrey = "#A6A6A6";
export const altoGrey = "#d9d9d9";
export const separatorGrey = "#393939";
export const inputBorder = "#BEBEBE";
export const inputPlaceholder = "#DADADA";
export const chartGrey = "#C4C4C4";

export const darkYellow = "#FABD59";
export const middleYellow = "#ECB832";
export const warningYellow = "#FECA43";
export const rgbaWhite = "rgba(255, 255, 255, 0.35)";

export const chartOrange = "#FF9C00";

export const alabaster = "#f7f7f7";

export const rgbaLightGrey = "rgba(0, 0, 0, 0.0970618)";
export const rgbaShadowGrey = "rgba(155, 155, 155, 0.04)";
export const rgbaStrongShadowGrey = "rgba(155, 155, 155, 0.3)";
export const rgbaShadowBlack = "rgba(0, 0, 0, 0.2)";
export const rgbaHeaderRow = "rgba(216, 216, 216, 0.1)";
export const rgbaMediumGrey = "rgba(190, 190, 190, 0.2)";
export const blackPointSeven = "rgba(0, 0, 0, 0.7)";

const azureRadiance = "#009cff";
const alto = "#DADADA";

export const tableHeader = "#F7F7F7";

export default {
  black,
  lightBlack,
  lighterBlack,
  white,
  darkWhite,
  seashell,
  red,
  darkRed,
  blue,
  purple,
  green,
  lighterGreen,
  lightGreen,
  darkenGrey,
  darkGreen,
  darkGrey,
  grey,
  ligthGrey,
  gainsboro,
  navBarGrey,
  ligtherGrey,
  darkYellow,
  rgbaLightGrey,
  rgbaShadowGrey,
  rgbaStrongShadowGrey,
  shark,
  dustyGray,
  disabledGrey,
  lightBlue,
  persianRed,
  mineShaft,
  altoGrey,
  wildWatermelon,
  separatorGrey,
  middleYellow,
  rgbaShadowBlack,
  inputPlaceholder,
  persianPlum,
  cherub,
  azalea,
  rgbaWhite,
  darkPurple,
  blackPointSeven,
  rgbaHeaderRow,
  rgbaMediumGrey,
  azureRadiance,
  alto,
  alabaster,
  tableHeader,
};
