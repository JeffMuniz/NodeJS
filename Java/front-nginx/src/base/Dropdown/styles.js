import styled from "styled-components";

import { inputBorder, blue, white } from "src/styles/colors";

export const DropdownContainer = styled.div`
  display: flex;
  justify-content: center;
`;

export const ActionWrapper = styled.div`
  cursor: pointer;
`;

export const OptionsContainer = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const SelectOptions = styled.ul`
  position: absolute;
  background: ${white};
  box-shadow: 1px 4px 20px rgba(0, 0, 0, 0.2);
  border-radius: 4px;
  padding: 0;
  margin-top: 40px;
  width: ${props => props.optionsWidth || "200px"};
  z-index: 5;
`;

export const StyledOption = styled.li`
  cursor: pointer;
  border-bottom: 1px solid ${inputBorder};
  color: ${blue};
  display: block;
  font-weight: 600;
  list-style: none;
  text-align: left;
  padding: 20px 0;
  margin: 0 25px;

  &:last-child {
    border-bottom: none;
  }
`;
