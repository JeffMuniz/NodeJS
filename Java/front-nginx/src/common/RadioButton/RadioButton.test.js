import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import TestRenderer from "react-test-renderer";

import { TextError, Box } from "./RadioButton.styles";
import RadioButton from "./RadioButton.js";

let defaultProps = {};
let testRenderer = null;
let instance = null;

const radioValues = [
  { value: true, label: "Sim", id: "Y" },
  { value: false, label: "NÃ£o", id: "N" },
];

beforeEach(() => {
  defaultProps = {
    options: radioValues,
    name: "radio",
    id: "radioId",
    onChange: jest.fn(),
    selectedValue: "",
    setFieldValue: jest.fn(),
    setFieldTouched: jest.fn(),
    values: { option: "" },
    labelBefore: true,
    isDisabled: false,
    error: null,
  };

  testRenderer = TestRenderer.create(<RadioButton {...defaultProps} />);
  instance = testRenderer.root;
});

describe("RadioButton", () => {
  it("should render it with default props", () => {
    // given
    const renderer = new ShallowRenderer();

    // when
    renderer.render(<RadioButton {...defaultProps} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });

  it("should render it when labelBefore is false", () => {
    // given
    const renderer = new ShallowRenderer();
    const labelBefore = false;

    // when
    renderer.render(
      <RadioButton {...defaultProps} labelBefore={labelBefore} />,
    );
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });

  it("should render Error when error is given", () => {
    // given
    const error = "error test";

    // when
    testRenderer.update(<RadioButton {...defaultProps} error={error} />);

    const errors = instance.findAllByType(TextError);

    // then
    expect(errors).toHaveLength(1);
  });

  it("should call onChange when button props change", () => {
    // given
    let calls = false;

    // when
    testRenderer.update(<RadioButton {...defaultProps} />);
    const boxes = instance.findAllByType(Box);

    boxes.map(box => {
      box.props.onChange("teste");

      // then
      expect(defaultProps.onChange).toBeCalled();
      calls = true;
      return null;
    });

    // then
    expect(calls).toBeTruthy();
  });

  it("should not rendering Radio Button when not pass options", () => {
    // when
    testRenderer.update(<RadioButton {...defaultProps} options={[]} />);

    const boxes = instance.findAllByType(Box);

    expect(boxes).toHaveLength(0);
  });
});
