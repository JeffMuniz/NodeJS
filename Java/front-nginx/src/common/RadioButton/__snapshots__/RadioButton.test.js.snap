// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should render it when labelBefore is false 1`] = `
<styled.div>
  <styled.label
    htmlFor="radioId_Y"
  >
    <styled.input
      checked={false}
      disabled={false}
      id="radioId_Y"
      name="radio"
      onBlur={[Function]}
      onChange={[Function]}
      type="radio"
      value=""
    />
    <styled.span />
    <styled.span
      labelBefore={false}
    >
      Sim
    </styled.span>
  </styled.label>
  <styled.label
    htmlFor="radioId_N"
  >
    <styled.input
      checked={false}
      disabled={false}
      id="radioId_N"
      name="radio"
      onBlur={[Function]}
      onChange={[Function]}
      type="radio"
      value=""
    />
    <styled.span />
    <styled.span
      labelBefore={false}
    >
      Não
    </styled.span>
  </styled.label>
</styled.div>
`;

exports[`RadioButton should render it with default props 1`] = `
<styled.div>
  <styled.label
    htmlFor="radioId_Y"
  >
    <styled.span
      labelBefore={true}
    >
      Sim
    </styled.span>
    <styled.input
      checked={false}
      disabled={false}
      id="radioId_Y"
      name="radio"
      onBlur={[Function]}
      onChange={[Function]}
      type="radio"
      value=""
    />
    <styled.span />
  </styled.label>
  <styled.label
    htmlFor="radioId_N"
  >
    <styled.span
      labelBefore={true}
    >
      Não
    </styled.span>
    <styled.input
      checked={false}
      disabled={false}
      id="radioId_N"
      name="radio"
      onBlur={[Function]}
      onChange={[Function]}
      type="radio"
      value=""
    />
    <styled.span />
  </styled.label>
</styled.div>
`;
